site/content/integrate/apps/functionality/external-webhooks/_index.md

```diff
@@ -1,178 +1,89 @@
-	],
-	"on_install": {
-		"path": "/install",
-		"expand": {
-			"admin_access_token": "all"
-		}
-	},
-	"http": {
-		"root_url": "http://localhost:8081"
-	}
-}
-```
+Using the above example values, the webhook URL will look like this:
 
-### Bindings and locations
+`http://my-mattermost-server/plugins/com.mattermost.apps/apps/my-apps/webhook/my-endpoint?secret=9a44ckeqytd3bftn3c3y53968o`
 
-The `Hello Webhooks!` app creates two commands: `/hello-webhooks info | send`.
+### Retrieve the secret key
 
-```json
-{
-    "type": "ok",
-    "data": [
-        {
-            "location": "/command",
-            "bindings": [
-                {
-                    "icon": "icon.png",
-                    "label": "hello-webhooks",
-                    "description": "Hello Webhooks App",
-                    "hint": "[ send ]",
-                    "bindings": [
-                        {
-                            "location": "send",
-                            "label": "send",
-                            "call": {
-                                "path": "/send"
-                            }
-                        },
-                        {
-                            "location": "info",
-                            "label": "info",
-                            "call": {
-                                "path": "/info"
-                            }
-                        }
-                    ]
-                }
-            ]
-        }
-    ]
-}
-```
+When the App is first installed, the `OnInstall` [call]({{<ref "/integrate/apps/structure/call">}}) handler will receive a context value of `app.webhook_secret` in the request. This value should be persisted by the App for authenticating incoming webhook requests.
 
-### Display the webhook command
+{{<note>}}
+The `OnInstall` call must have the `app` expand field set to `summary` for the generated secret key to be populated in the call request.
+{{</note>}}
 
-`/hello-webhooks info` displays a valid `/hello-webhooks send` command constructed with your mattermost-site-url, the path to the hello app webhook endpoint, and a secret.
+For example, an Golang App's [manifest]({{<ref "/integrate/apps/structure/manifest">}}) would define an `OnInstall` call to get the secret like this:
 
 ```go
-func info(w http.ResponseWriter, req *http.Request) {
-    creq := apps.CallRequest{}
-    json.NewDecoder(req.Body).Decode(&creq)
-
-    json.NewEncoder(w).Encode(apps.CallResponse{
-        Markdown: md.Markdownf("Try `/hello-webhooks send %s`",
-            creq.Context.MattermostSiteURL+creq.Context.AppPath+apps.PathWebhook+
-                "/hello"+
-                "?secret="+creq.Context.App.WebhookSecret),
-    })
+appManifest = apps.Manifest{
+    AppID:       apps.AppID("hello-world"),
+    Version:     apps.AppVersion("0.1.0"),
+    HomepageURL: "https://my-site/my-repo",
+    DisplayName: "Hello, world!",
+    RequestedPermissions: apps.Permissions{
+        apps.PermissionActAsBot,
```

Comment:
Need the webhook permission? See https://github.com/mattermost/mattermost-app-examples/blob/master/golang/webhooks/hello.go#L34

Reply:
Great catch! Fixed. :)